sympy/diffgeom/diffgeom.py

```diff
@@ -1154,7 +1171,31 @@ def basishelper(p,k):
                 else:
                     indices.remove(p)
         self.tensor = dict(zip(indices,coeffs))
-    def init_from_array(self,component_array,variance,coordinate_system):
+    def to_NDimArray(self):
+        """
+        Converts from internal sparse dict representation to sympy.tensor.MutableSparseNDimArray
```

Comment:
SparseNDimArray also has dict. Would it be possible to use the same representation with no extra conversion?

Reply:
SparseNDimArray uses int keys and reshaping.  TensorArray dicts are keyed with tuples.  This makes it relatively straightforward to implement operations like tensor product and contraction which, as far as I can tell, are either not implemented or experimental in the tensor classes.

Reply:
Does that also mean that none of the methods of NDimArray classes (sparse or dense) can be used for TensorArray?

Reply:
It may be possible to write a wrapper, but currently that is true.